src/Challenge.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import './style/Challenge.css';
+import './style/styles.css';
+import Button from 'react-button-component'
+import Header from './components/header'
+import DifficultyTab from './DifficultyTab'
+import ShowSpinner from './components/spinner';
+
+import { 
+  fetchAllChapterNames, 
+  fetchAllPairs, 
+  fetchCompletedLessons,
+  fetchLessonById
+} from './actions/learn'
+
+import { getCurrentLessonForUser } from './actions/homepage';
+
+const CustomButton = Button.extend`
+    margin-top:4vh;
+    height: 82px;	
+    width: 270px;	
+    border: 5px solid #F5A623;	
+    border-radius: 10px;	
+    background-color: #FFFFFF;
+    font-size:30px;
+
+`
+
+
+class Challenge extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { 
+      headerLinks: ["Games", "Learn", "Home"],
+      isEzActive:false,
+      isMidActive:false,
+      isHardActive: false
+    }
+    this.BeginChallenge=this.BeginChallenge.bind(this)
+    this.EzSelected= this.EzSelected.bind(this)
+    this.MedSelected=this.MedSelected.bind(this)
+    this.HardSelected=this.HardSelected.bind(this)
+  }
+
+  EzSelected()
+  {
+    this.setState({
+      isEzActive:true,
+      isMidActive:false,
+      isHardActive: false
+    })
+    console.log("difficulty selected:easy")
+  }
+
+  MedSelected()
+  {
+    this.setState({
+      isEzActive:false,
+      isMidActive:true,
+      isHardActive: false
+    })
+    console.log("difficulty selected: Medium ")
+  }
+
+  HardSelected()
+  {
+    this.setState({
+      isEzActive:false,
+      isMidActive:false,
+      isHardActive: true
+    })
+    console.log("difficulty selected: Hard" )
+  }
+  BeginChallenge(state){
+    var difficulty="";
```

Comment:
dont use `var`, always prefer to use const if you intend the variable to be immutable or `let` if its going to be mutable